### PR TITLE
Feature stock entry report currency

### LINKED ISSUE #5701: Stock Entry Report should allow you to set a currency
In keeping with the recent change to the expiration report (#5698) and stock exit report (#5693), we should allow users to pick a currency on the stock exit report.  In principle, all reports should provide the option to switch currencies on the fly.  The currency select would be one of the options below:

image

### PR DESCRIPTION
**Pull request by lomamech**
- Set currency in stock entry report
- Udpate Entry report display currency value

closes #5701

### CODE REVIEW

**Comment by lomamech** [comment]
image

File: client/src/modules/reports/generate/stock_entry/stock_entry.html [hunk 1]
```diff
@@ -39,6 +39,11 @@ <h3 translate>REPORT.STOCK.ENTRY_REPORT</h3>
             required="true">
           </bh-date-interval>
 
+          <bh-currency-select
+            currency-id="ReportConfigCtrl.currency_id"
```
Comment (jniles) [inline client/src/modules/reports/generate/stock_entry/stock_entry.html]: I think this needs to be `ReportConfigCtrl.reportDetails.currencyId`.  That way, it matches the saved value, and doesn't reset to the enterprise currency each time.

File: client/src/modules/reports/generate/stock_entry/stock_entry.config.js [hunk 2]
```diff
@@ -22,6 +22,8 @@ function StockEntryConfigController($sce, Notify, SavedReports, AppCache, report
   vm.previewGenerated = false;
   vm.onEntryTypeChange = onEntryTypeChange;
 
+  vm.currency_id = Session.enterprise.currency_id;
```
Comment (jniles) [inline client/src/modules/reports/generate/stock_entry/stock_entry.config.js]: Can you set this to 
```js
vm.reportDetails.currencyId = Session.enterprise.currency_id;
```
instead?  That way, the component is preloaded and can be overridden by a cached currency.

File: server/controllers/stock/reports/stock/entry_report.js [hunk 3]
```diff
@@ -27,87 +27,103 @@ function stockEntryReport(req, res, next) {
...
   const collection = {};
 
-  // entry to purchase
+  // entryFromPurchase: [], entryFromIntegration: [], entryFromDonation: [], entryFromTransfer
+
+  // get stock entry from purchase
   collection.entryFromPurchase = formatAndCombine(entryCollection.entryFromPurchase);
 
-  // entry to integration
+  // get stock entry from integration
```
Comment (jniles) [inline server/controllers/stock/reports/stock/entry_report.js:74]: 👍

**Review by jniles — COMMENTED** [review]
Code looks pretty good.  I haven't tested this on the live server, but just had some quick feedback about the `reportDetails` cache.

**Review by jniles — APPROVED** [review]
Looks good to me!

bors r+

**Pull request merged**